yt/data_objects/index_subobjects/octree_subset.py

```diff
@@ -43,13 +43,10 @@ class OctreeSubset(YTSelectionContainer):
     _cell_count = -1
     _block_order = "C"
 
-    def __init__(
-        self, base_region, domain, ds, over_refine_factor=1, num_ghost_zones=0
-    ):
+    def __init__(self, base_region, domain, ds, num_zones=2, num_ghost_zones=0):
```

Comment:
@BolunThompson Could you add a shim here, so that if `over_refine_factor` is specified it sets `num_zones` appropriately (and emits a deprecation warning)?  We can disable the behavior in 4.2.